### ISSUE TITLE
About model's output

### ISSUE THREAD

**Issue by ShallTearchen** [kmkurn/pytorch-crf#49]
Can the output of the model be directly propagated as a loss?

**Comment by kmkurn** [comment]
By output you mean the output of `decode`? If so, then it depends on the loss function. `decode` is not differentiable w.r.t to the model's parameters. So, if you require the loss function to be differentiable (which normally you do since you're using pytorch), then no.

**Comment by ShallTearchen** [comment]
@kmkurn OK thx!
